community/kernel/src/main/java/org/neo4j/kernel/impl/index/labelscan/LabelScanLayout.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.index.labelscan;
+
+import org.neo4j.index.gbptree.GBPTree;
+import org.neo4j.index.gbptree.Layout;
+import org.neo4j.io.pagecache.PageCursor;
+
+import static java.lang.Long.bitCount;
+
+/**
+ * {@link Layout} for {@link GBPTree} used by {@link NativeLabelScanStore}.
+ *
+ * <ul>
+ * <li>
+ * Each keys is a combination of {@code labelId} and {@code nodeIdRange} ({@code nodeId/64}).
+ * </li>
+ * <li>
+ * Each value is a 64-bit bit set (a primitive {@code long}) where each set bit in it represents
+ * a node with that label, such that {@code nodeId = nodeIdRange+bitOffset}. Range size (e.g. 64 bits)
+ * is configurable on initial creation of the store, 8, 16, 32 or 64.
+ * </li>
+ * </ul>
+ */
+class LabelScanLayout implements Layout<LabelScanKey,LabelScanValue>
+{
+    /**
+     * Name part of the {@link #identifier()} value.
+     */
+    private static final String IDENTIFIER_NAME = "LSL";
+
+    /**
+     * Size of each {@link LabelScanKey}.
+     */
+    private static final int KEY_SIZE = Integer.BYTES/*labelId*/ + 6/*idRange*/;
+
+    /**
+     * Size of each node id range, e.g. 8, 16, 32 or 64. This value is written at creation,
+     * otherwise verified against index meta data on open.
+     */
+    private final int rangeSize;
+
+    /**
+     * {@link #rangeSize}, as number of bytes (instead of number of bits).
+     */
+    private final int rangeSizeBytes;
+
+    LabelScanLayout( int rangeSize )
```

Comment:
Initially it was there so that we could compare different range sizes against each other, although I'd guess landing at the 64-bit variant for the Lucene LSS tells the tale that this investigation has already been done and we could simply just do that.

We almost removed rangeSize as part of this PR, but narrowly decided to keep it. This may have tipped the scale. Removing it.